### PR TITLE
#63 modify extract

### LINKED ISSUE #63: Modify Extract
**Description**
Modify `extract.py` to read from the newly laid out S3 Bucket.

**Outcome**
Extracts Dataframe from S3 Bucket which can be passed onto `transform.py`

**User story**
As a data engineer,
I want `extract.py` to read data from the newly structured S3 bucket,
so that the extracted DataFrame can be seamlessly passed to `transform.py` for processing.

### PR DESCRIPTION
**Pull request by keoghrmj**
Extract (and transform) have been modified to extract all necessary information from the S3 Bucket.

Closes #63

### CODE REVIEW

File: pipeline/etl.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import pandas as pd
+from extract import main as emain
+from transform import main as tmain
+
+def main(topic: list[str]) -> pd.DataFrame:
```
Comment (SurinaCS) [inline pipeline/etl.py:5]: needs a module docstring please!

File: pipeline/extract.py [hunk 2]
```diff
@@ -42,95 +38,114 @@ def s3_connection() -> client:
             raise
         s3 = client(
             "s3",
-            aws_access_key_id=os.environ.get("AWS_ACCESS_KEY_ID"),
-            aws_secret_access_key=os.environ.get(
-                "AWS_SECRET_ACCESS_KEY"),
+            aws_access_key_id=aws_access_key,
+            aws_secret_access_key=aws_secret_key,
```
Comment (SurinaCS) [inline pipeline/extract.py:44]: good correction

File: pipeline/extract.py [hunk 3]
```diff
@@ -42,95 +40,115 @@ def s3_connection() -> client:
...
             config=config
         )
-    except Exception as e:
+    except ConnectionError as e:
```
Comment (SurinaCS) [inline pipeline/extract.py:47]: nice specific error!

File: pipeline/extract.py [hunk 4]
```diff
@@ -42,95 +40,115 @@ def s3_connection() -> client:
...
+            count = file_content.count(keyword)
+            if count > 0:
+                logging.info("Keyword %s found in %s %d times", keyword, file_name, count)
+                sentiment_scores[keyword] = add_sentiment_scores(file_content)
+                keyword_counts[keyword] += count
+
+        return {
+            "Hour": hour_folder,
+            "Counts": dict(keyword_counts),
+            "Sentiment Score" : sentiment_scores
```
Comment (SurinaCS) [inline pipeline/extract.py:101]: nice

File: pipeline/etl.py (hunk 1 above)
Comment (SurinaCS) [reply pipeline/etl.py:5]:
```suggestion
def main(topic: list[str]) -> pd.DataFrame:
"""Main function to extract and transform the data from BlueSky and Google trends"
```

**Review by SurinaCS — CHANGES_REQUESTED** [review]
great work, just needing a module docstring

File: pipeline/etl.py (hunk 1 above)
Comment (keoghrmj) [reply pipeline/etl.py:5]: This was just a test file to run extract and transform back to back. It will be created again once load has been finished

**Review by SurinaCS — APPROVED** [review]

**Pull request merged**